### PR TITLE
custom client transport should be a constructor (IMO)

### PR DESCRIPTION
**Pull request by amitport**
Hi,

I'm implementing a custom transport. Having a parameter that just override transport is not very useful as the transport will need access to things like the store (same way that local and socketio transport do).

#### Checklist

* [V] Use a separate branch in your local repo (not `master`).
* [V] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

**Comment by nicolodavis** [comment]
Just curious, what transport are you implementing?

**Comment by amitport** [comment]
few things:

1. higher level transport a-la https://github.com/primus/primus
2. mirror-like transport for integeration with my own https://github.com/CardForest/master-class (which is like a proxy based state-machine master that is transperantly local or remote :) (also used instead of redux))

**Comment by nicolodavis** [comment]
Sounds good.

Can you update the test in client.test.js that is now failing?

**Comment by amitport** [comment]
@nicolodavis 
changed so transport is now nested under `multiplayer.server`. Makes sense to me since transport is a property of the server config. The entire `multiplayer.server` is passed to the transport and may contain more custom configuration.

BTW I think there is room for bigger refactor here (I wrote to you in in gitter: 

may want to lift entire multiplayer/transport selection logic outside of `_ClientImpl`, enable cleaner config and smaller bundles for those not using socketIO

)

File: src/client/client.js [hunk 1]
```diff
@@ -272,8 +272,6 @@ class _ClientImpl {
           server: multiplayer.server,
           socketOpts,
         });
-      } else if (multiplayer.transport !== undefined) {
-        this.transport = multiplayer.transport;
```
Comment (nicolodavis) [inline src/client/client.js]: Shall we just leave this option as `multiplayer.transport` and treat it as a constructor here?  I'd prefer to use `server` for the socketio stuff.

Basically I'm proposing changing this line to `this.transport = new (multiplayer.transport)({ ... })` and pass in everything that's passed in from the if branch right above.

**Pull request merged**